homeassistant/components/sensor/coinmarketcap.py

```diff
@@ -13,54 +13,70 @@
 import homeassistant.helpers.config_validation as cv
 from homeassistant.components.sensor import PLATFORM_SCHEMA
 from homeassistant.const import (
-    ATTR_ATTRIBUTION, CONF_CURRENCY, CONF_DISPLAY_CURRENCY)
+    ATTR_ATTRIBUTION, CONF_CURRENCY_ID, CONF_DISPLAY_CURRENCY,
+    CONF_DISPLAY_CURRENCY_DECIMALS)
 from homeassistant.helpers.entity import Entity
 
-REQUIREMENTS = ['coinmarketcap==4.2.1']
+REQUIREMENTS = ['coinmarketcap==5.0.3']
 
 _LOGGER = logging.getLogger(__name__)
 
-ATTR_24H_VOLUME = '24h_volume'
+ATTR_VOLUME_24H = 'volume_24h'
 ATTR_AVAILABLE_SUPPLY = 'available_supply'
+ATTR_CIRCULATING_SUPPLY = 'circulating_supply'
 ATTR_MARKET_CAP = 'market_cap'
 ATTR_PERCENT_CHANGE_24H = 'percent_change_24h'
 ATTR_PERCENT_CHANGE_7D = 'percent_change_7d'
 ATTR_PERCENT_CHANGE_1H = 'percent_change_1h'
 ATTR_PRICE = 'price'
+ATTR_RANK = 'rank'
 ATTR_SYMBOL = 'symbol'
 ATTR_TOTAL_SUPPLY = 'total_supply'
 
 CONF_ATTRIBUTION = "Data provided by CoinMarketCap"
 
-DEFAULT_CURRENCY = 'bitcoin'
+DEFAULT_CURRENCY_ID = 1
 DEFAULT_DISPLAY_CURRENCY = 'USD'
+DEFAULT_DISPLAY_CURRENCY_DECIMALS = 2
 
 ICON = 'mdi:currency-usd'
 
 SCAN_INTERVAL = timedelta(minutes=15)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_CURRENCY, default=DEFAULT_CURRENCY): cv.string,
+    vol.Optional(CONF_CURRENCY_ID, default=DEFAULT_CURRENCY_ID):
+        cv.positive_int,
     vol.Optional(CONF_DISPLAY_CURRENCY, default=DEFAULT_DISPLAY_CURRENCY):
         cv.string,
+    vol.Optional(CONF_DISPLAY_CURRENCY_DECIMALS,
+                 default=DEFAULT_DISPLAY_CURRENCY_DECIMALS):
+        cv.positive_int,
 })
 
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the CoinMarketCap sensor."""
-    currency = config.get(CONF_CURRENCY)
-    display_currency = config.get(CONF_DISPLAY_CURRENCY).lower()
+    currency_id = int(config.get(CONF_CURRENCY_ID))
+    display_currency = config.get(CONF_DISPLAY_CURRENCY).upper()
+    display_currency_decimals = int(config.get(CONF_DISPLAY_CURRENCY_DECIMALS))
 
     try:
-        CoinMarketCapData(currency, display_currency).update()
+        CoinMarketCapData(
+            currency_id, display_currency, display_currency_decimals
+        ).update()
     except HTTPError:
-        _LOGGER.warning("Currency %s or display currency %s is not available. "
-                        "Using bitcoin and USD.", currency, display_currency)
-        currency = DEFAULT_CURRENCY
+        _LOGGER.warning("Currency ID %d, display currency %s or "
+                        "rounding off %d decimals is not available. "
+                        "Using bitcoin and USD.",
+                        currency_id, display_currency,
+                        display_currency_decimals)
+        currency_id = DEFAULT_CURRENCY_ID
         display_currency = DEFAULT_DISPLAY_CURRENCY
+        display_currency_decimals = DEFAULT_DISPLAY_CURRENCY_DECIMALS
```

Comment:
`display_currency_decimals` doesn't causes `HTTPError`s. Please remove all references to `display_currency_decimals` from the log entry. Also, there is no need to set the default value again.